transformer_lens/components.py

```diff
@@ -640,12 +637,12 @@ def apply_causal_mask(
 
     def rotary_rotate_qk(
         self,
-        q: Float[torch.Tensor, "batch pos head_index d_head"],
-        k: Float[torch.Tensor, "batch pos head_index d_head"],
+        q: Float[torch.Tensor, "batch q_pos head_index d_head"],
+        k: Float[torch.Tensor, "batch k_pos head_index d_head"],
         past_kv_pos_offset,
     ) -> Tuple[
-        Float[torch.Tensor, "batch pos head_index d_head"],
-        Float[torch.Tensor, "batch pos head_index d_head"],
+        Float[torch.Tensor, "batch q_pos head_index d_head"],
+        Float[torch.Tensor, "batch k_pos head_index d_head"],
     ]:
```

Comment:
@alan-cooney @jbloomAus I had to change this typehint to get one of the tests to pass (test_hooked_transformer::test_dtypes). It failed because this method got passed `q`, `k` with different `pos` dimensions and jaxtyping checks for that as configured.  I'm not familiar enough with this method and its usage to say whether `q`, `k` should be allowed to have different `pos` dimensions.

Reply:
Are you sure it's actually getting passed different sized queries and keys, or are the defined type signatures at this point just different?

I think this may be an issue with Beartype getting confused about the size of one of these tensors at a previous point (perhaps in the `if past_kv_cache_entry is not None:` section)?

I'm not very familiar with rotary embeddings either but as far as I can tell the q and k passed in here should be the same size (and they should also be the same size afterwards as they're immediately dot producted).

Reply:
> Are you sure it's actually getting passed different sized queries and keys, or are the defined type signatures at this point just different?

Yep, I confirmed that they were different sizes.  

> I think this may be an issue with Beartype getting confused about the size of one of these tensors at a previous point (perhaps in the if past_kv_cache_entry is not None: section)?

Hmm, I don't think this is the case.  Beartype/Jaxtyping should only look at the arguments passed to the call in this case

Here's a minimal script to reproduce (you'll have to change the typehints back):
```python
from jaxtyping import install_import_hook
with install_import_hook("transformer_lens", "beartype.beartype"):
    from transformer_lens import HookedTransformer
    model = HookedTransformer.from_pretrained("EleutherAI/pythia-70m")
    _ = model.generate("Hello, World!")
``` 

Actually, reading the code more, I think my change is correct and we should allow different sizes in the `pos` dim.  The reason is that when we're generating completion, after we feed through the initial input, we're using only the latest token as the query vec and using the kv cache for the key vecs to compute attention scores.  It's ok for `q` and `k` to have different sizes in the `pos` dim because they are getting dotted over the `d_head` dim.  Finally, the `past_kv_pos_offset` is just so we get the right positional embedding for `q`.